src/libstd/collections/hash/map.rs

```diff
@@ -503,7 +498,7 @@ fn robin_hood<'a, K: 'a, V: 'a>(bucket: FullBucketMut<'a, K, V>,
         loop {
             displacement += 1;
             let probe = bucket.next();
-            debug_assert!(probe.index() != idx_end);
+            debug_assert!(probe.index() != start_index);
```

Comment:
Is this invariant not as strict as the old one? Either way it's ok. You could add a comment that this assertion is simplified.